frontend/src/services/mock_backend/mock_your_clubs_backend.ts

```diff
@@ -0,0 +1,52 @@
+import { BOOKS_LIST } from "../../utils/mock_club_data";
+import { CLUB_DESCRIPTIONS } from "../../utils/mock_club_data";
+import { CLUB_NAMES } from "../../utils/mock_club_data";
+import { CONTENT_WARNINGS } from "../../utils/mock_club_data";
+import { MAX_NUM_CLUBS } from "../../utils/temporary_testing_consts";
+
+export interface BookProps {
+  title: string, 
+  author: string, 
+  isbn: string, 
+}
+
+export interface ClubProps {
+  name: string;
+  description: string;
+  contentWarnings: Array<string>; 
+  currentBook: BookProps,
+}
+
+/**
+ * Mimics the functionality of a Java Servlet that fetches list of clubs
+ * that the user is in.
+ */
+export class MockYourClubsBackendService {
+  private mockClubs: ClubProps[] = [];
+
+  constructor() {
+    for (let i = 0; i < MAX_NUM_CLUBS; i++) {
+      this.mockClubs.push({
+        name: CLUB_NAMES[i],
+        description: CLUB_DESCRIPTIONS[i],
+        contentWarnings: CONTENT_WARNINGS[i],
+        currentBook: BOOKS_LIST[i],
+      })
+    }
+  }
+
+  /** TODO: Update parameters to include Membership status, user Id, page. */
+  listClubs(numClubs: number): Promise<string> {
+    let listedClubs: ClubProps[] = [];
+    for (let i = 0; i < Math.min(numClubs, MAX_NUM_CLUBS); i++) {
```

Comment:
nit: Directly use `this.mockClubs.length` instead of `NUM_MAX_CLUBS`. I know they're supposed to be the same but something could potentially go wrong (or change) in the constructor and lead to the actual list being shorter than the max.